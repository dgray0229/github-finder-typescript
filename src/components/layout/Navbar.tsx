import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faGithub } from '@fortawesome/free-brands-svg-icons'
// import PropTypes from 'prop-types'

// We define the type of our props in a separate object
type NavbarProps = {
    title: string,
    // The question mark means that the icon property is optional
    icon?: any,
}

// We then pass that type to our navbar component
const Navbar = ( {title}: NavbarProps) => {
    // Beacuse this is a funcitonal component, we no longer define default props inside of the component

    return (
        <header>
            <nav className="navbar bg-primary">
                <h1><FontAwesomeIcon icon={faGithub} /> {title!}</h1>
            </nav>
        </header>
    )

}
// Here is where we define default props for the navbar
Navbar.defaultProps = {
    // We assert that this property is non-null below with our '!' non-null assertion operator
    title: "Github Finder"
};


export default Navbar
