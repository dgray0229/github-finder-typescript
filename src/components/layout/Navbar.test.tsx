import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Navbar from './Navbar';

Enzyme.configure({ adapter: new Adapter() })

describe("Navbar Component", () => {
  test("renders main component", () => {
    const wrapper = shallow(<Navbar />)
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot();

  })
})
