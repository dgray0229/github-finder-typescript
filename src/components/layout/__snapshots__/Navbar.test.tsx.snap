// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component renders main component 1`] = `ShallowWrapper {}`;
